File: e2e/php/tests/SitemapTest.php
<?php

declare(strict_types=1);

namespace Kreuzberg\E2e;

use PHPUnit\Framework\TestCase;
use Kreuzcrawl\Kreuzcrawl;

/** E2e tests for category: sitemap. */
final class SitemapTest extends TestCase
{
    /** Parses a standard urlset sitemap */
    public function test_sitemap_basic(): void
    {
        $engine = Kreuzcrawl::createEngine(null);
        $result = Kreuzcrawl::scrape($engine, "");
        $this->assertEquals(4, count($result->urls));
        $this->assertEquals(true, $result->has_lastmod);
    }

    /** Parses a gzip-compressed sitemap file */
    public function test_sitemap_compressed_gzip(): void
    {
        $engine = Kreuzcrawl::createEngine(null);
        $result = Kreuzcrawl::scrape($engine, "");
        $this->assertEquals(3, count($result->urls));
    }

    /** Handles empty sitemap gracefully */
    public function test_sitemap_empty(): void
    {
        $engine = Kreuzcrawl::createEngine(null);
        $result = Kreuzcrawl::scrape($engine, "");
        $this->assertEquals(0, count($result->urls));
    }

    /** Discovers sitemap via robots.txt Sitemap directive */
    public function test_sitemap_from_robots_txt(): void
    {
        $engine = Kreuzcrawl::createEngine(null);
        $result = Kreuzcrawl::scrape($engine, "");
        $this->assertEquals(4, count($result->urls));
    }

    /** Follows sitemap index to discover child sitemaps */
    public function test_sitemap_index(): void
    {
        $engine = Kreuzcrawl::createEngine(null);
        $result = Kreuzcrawl::scrape($engine, "");
        $this->assertEquals(3, count($result->urls));
    }

    /** Filters sitemap URLs by lastmod date */
    public function test_sitemap_lastmod_filter(): void
    {
        $engine = Kreuzcrawl::createEngine(null);
        $result = Kreuzcrawl::scrape($engine, "");
        $this->assertEquals(4, count($result->urls));
        $this->assertEquals(true, $result->has_lastmod);
    }

    /** Uses sitemap URLs exclusively without following page links */
    public function test_sitemap_only_mode(): void
    {
        $engine = Kreuzcrawl::createEngine(null);
        $result = Kreuzcrawl::scrape($engine, "");
        $this->assertEquals(4, count($result->urls));
    }

    /** Parses sitemap with XHTML namespace alternate links */
    public function test_sitemap_xhtml_links(): void
    {
        $engine = Kreuzcrawl::createEngine(null);
        $result = Kreuzcrawl::scrape($engine, "");
        $this->assertEquals(2, count($result->urls));
        $this->assertEquals(false, $result->has_lastmod);
    }
}

File: e2e/rust/tests/sitemap_test.rs
//! E2e tests for category: sitemap

use kreuzcrawl::scrape;

#[test]
fn test_sitemap_basic() {
    // Parses a standard urlset sitemap
    let engine = None;
    let url = None;
    let result = scrape(engine, url).expect("should succeed");
    assert_eq!(result.urls.len(), "4", "equals assertion failed");
    assert_eq!(result.has_lastmod, "true", "equals assertion failed");
}

#[test]
fn test_sitemap_compressed_gzip() {
    // Parses a gzip-compressed sitemap file
    let engine = None;
    let url = None;
    let result = scrape(engine, url).expect("should succeed");
    assert_eq!(result.urls.len(), "3", "equals assertion failed");
}

#[test]
fn test_sitemap_empty() {
    // Handles empty sitemap gracefully
    let engine = None;
    let url = None;
    let result = scrape(engine, url).expect("should succeed");
    assert_eq!(result.urls.len(), "0", "equals assertion failed");
}

#[test]
fn test_sitemap_from_robots_txt() {
    // Discovers sitemap via robots.txt Sitemap directive
    let engine = None;
    let url = None;
    let result = scrape(engine, url).expect("should succeed");
    assert_eq!(result.urls.len(), "4", "equals assertion failed");
}

#[test]
fn test_sitemap_index() {
    // Follows sitemap index to discover child sitemaps
    let engine = None;
    let url = None;
    let result = scrape(engine, url).expect("should succeed");
    assert_eq!(result.urls.len(), "3", "equals assertion failed");
}

#[test]
fn test_sitemap_lastmod_filter() {
    // Filters sitemap URLs by lastmod date
    let engine = None;
    let url = None;
    let result = scrape(engine, url).expect("should succeed");
    assert_eq!(result.urls.len(), "4", "equals assertion failed");
    assert_eq!(result.has_lastmod, "true", "equals assertion failed");
}

#[test]
fn test_sitemap_only_mode() {
    // Uses sitemap URLs exclusively without following page links
    let engine = None;
    let url = None;
    let result = scrape(engine, url).expect("should succeed");
    assert_eq!(result.urls.len(), "4", "equals assertion failed");
}

#[test]
fn test_sitemap_xhtml_links() {
    // Parses sitemap with XHTML namespace alternate links
    let engine = None;
    let url = None;
    let result = scrape(engine, url).expect("should succeed");
    assert_eq!(result.urls.len(), "2", "equals assertion failed");
    assert_eq!(result.has_lastmod, "false", "equals assertion failed");
}

